### PR TITLE
Use assimp to load stl

### LINKED ISSUE #712: support ascii stl files
I'm using rviz installed from ubuntu 20.04 binaries at foxy, trying to load a robot model exported from openscad.

I'm getting an error with:
`... rviz can only load binary STL files so it will not be loaded. Please convert it to a binary STL file.`
similar to rviz https://github.com/ros-visualization/rviz/issues/1364
which apparently is fixed in https://github.com/ros-visualization/rviz/pull/1365

### PR DESCRIPTION
**Pull request by ahcorde**
Related with this issue https://github.com/ros2/rviz/issues/712, this is the current behaviour of the ROS 1 version https://github.com/ros-visualization/rviz/pull/1365/files#diff-931ce7a05c370df33e2308f9adb590d3bde08241cf0e03dc86037731e3a5d83f

### CODE REVIEW

**Review by clalancette — COMMENTED** [review]
I only have one concern with this PR, which is whether all of the STL files we used to be able to load still load after this.  There are like 600 in the Rolling sources right now, which is obviously too many to test by hand.

I'm not quite sure how to test this; @ahcorde what's your feeling here?

**Comment by ahcorde** [comment]
> I only have one concern with this PR, which is whether all of the STL files we used to be able to load still load after this. There are like 600 in the Rolling sources right now, which is obviously too many to test by hand.
> 
> I'm not quite sure how to test this; @ahcorde what's your feeling here?

This behaviour is already working on ROS 1. Just to avoid discrepancies between them when porting to ROS 2.

**Comment by ahcorde** [comment]
friendly ping @clalancette

File: rviz_rendering_tests/test/mesh_loader_test.cpp [hunk 1]
```diff
@@ -135,13 +141,13 @@ TEST_F(MeshLoaderTestFixture, loading_invalid_stl_files_should_fail) {
   ASSERT_FALSE(rviz_rendering::loadMeshFromResource(mesh_path));
 }
 
-TEST_F(MeshLoaderTestFixture, loading_almost_valid_stl_files_should_succed) {
+TEST_F(MeshLoaderTestFixture, loading_almost_valid_stl_files_should_fail) {
   /// Load a "potentially" valid STL binary file with bigger size than the
   /// expected. The extra "unexpected" data at the end of the file should be
   /// ignored.
```
Comment (clalancette) [inline rviz_rendering_tests/test/mesh_loader_test.cpp]: I think we should remove this comment now, right?  That is, we will *not* succeed on these STL files anymore, if I understand the change.

File: rviz_rendering_tests/test/mesh_loader_test.cpp [hunk 2]
```diff
@@ -126,6 +125,13 @@ TEST_F(MeshLoaderTestFixture, loading_invalid_stl_files_fail) {
   ASSERT_FALSE(rviz_rendering::loadMeshFromResource(mesh_path));
 }
 
+TEST_F(MeshLoaderTestFixture, loading_valid_ascii_stl_file) {
```
Comment (clalancette) [inline rviz_rendering_tests/test/mesh_loader_test.cpp]: Isn't this actually loading an "invalid" ascii STL file now?  That is, we are changing this from valid -> invalid, so we should rename this (and probably the filename as well).

File: rviz_rendering_tests/test/mesh_loader_test.cpp [hunk 3]
```diff
@@ -135,13 +141,13 @@ TEST_F(MeshLoaderTestFixture, loading_invalid_stl_files_should_fail) {
   ASSERT_FALSE(rviz_rendering::loadMeshFromResource(mesh_path));
 }
 
-TEST_F(MeshLoaderTestFixture, loading_almost_valid_stl_files_should_succed) {
+TEST_F(MeshLoaderTestFixture, loading_almost_valid_stl_files_should_fail) {
   /// Load a "potentially" valid STL binary file with bigger size than the
   /// expected. The extra "unexpected" data at the end of the file should be
   /// ignored.
   std::string mesh_path = "package://rviz_rendering_tests/test_meshes/valid_extra.stl";
```
Comment (clalancette) [inline rviz_rendering_tests/test/mesh_loader_test.cpp]: Same comment here; I think we should probably rename this to `invalid_extra.stl`.

**Review by clalancette — CHANGES_REQUESTED** [review]
I've left a few minor comments on things to cleanup.

I have to say that while I think this is the correct thing to do, I'm a little nervous about it.  It seems like we'll be rejecting more STL files now.  But I guess the best we can do is try it out and see what happens.

**Comment by ahcorde** [comment]
We are rejecting files which are not well-formed and this is the current behaviour of ROS 1.

**Review by clalancette — APPROVED** [review]
Thanks for the changes!  This looks good to me with green CI.

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19660/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14167/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20383/)

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19665/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14172/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20388/)

**Pull request merged**